ultravox/tools/tts_tool.py

```diff
@@ -0,0 +1,73 @@
+import argparse
+import os
+
+import datasets
+
+from ultravox.tools import tts
+
+# This script is used to generate audio samples from text using a TTS model.
+# Ex: python tts_tool.py -d google/boolq -c question -a audio -u "fixie-ai/boolq-audio
+parser = argparse.ArgumentParser()
+parser.add_argument("--dataset-name", "-d", type=str, required=True)
+parser.add_argument("--dataset-subset", "-S", type=str)
+parser.add_argument("--dataset-split", "-s", type=str)
+parser.add_argument("--column-name", "-c", type=str, default="question")
+parser.add_argument("--audio-column-name", "-a", type=str)
+parser.add_argument("--num-samples", "-n", type=int)
+parser.add_argument("--voice", "-V", type=str)
+parser.add_argument("--sample-rate", "-r", type=int, default=16000)
+parser.add_argument("--upload-name", "-u", type=str)
+parser.add_argument("--token", "-t", type=str)
+
+
+def _tts_split(
+    tts_client: tts.AzureTts,
+    ds_split: datasets.IterableDataset,
+    col_name: str,
+    audio_col_name: str,
+):
+    def _tts_batch(batch):
+        batch[audio_col_name] = [
+            {"bytes": tts_client.tts(text)} for text in batch[col_name]
+        ]
+        return batch
+
+    return ds_split.map(_tts_batch, batched=True).cast_column(
```

Comment:
nit: [default `batch_size`](https://huggingface.co/docs/datasets/v2.19.0/en/package_reference/main_classes#datasets.Dataset.map) seems to be 1000, right? Or did I read that wrong?

I guess things are fine and there aren't too many requests if it worked?

Reply:
The requests are currently all serialized since we're not using asyncio or any multiprocessing for the TTS requests, so the batch size doesn't matter all that much. But it would be useful to rework this in a follow up PR, we currently can only process ~10 rows per second with the current approach.